apps/shared/y4m.c

```diff
@@ -380,6 +416,12 @@ avifBool y4mRead(const char * inputFilename, avifImage * avif, avifAppSourceTimi
         }
     }
 
+    // libavif API does not guarantee the absence of undefined behavior if samples exceed the specified avif->depth.
+    // Avoid that by making sure input values are within the correct range.
+    if (y4mClampSamples(avif)) {
+        fprintf(stderr, "WARNING: some samples were clamped to fit into %u bits per channel per sample\n", avif->depth);
```

Comment:
Nit: I think "per channel" could be omitted because "sample" generally means a single number. It's the word "pixel" that can mean multiple numbers (one per channel).